scrapyd/utils.py

```diff
@@ -154,3 +155,9 @@ def _to_native_str(text, encoding='utf-8', errors='strict'):
         return text.encode(encoding, errors)
     else:
         return text.decode(encoding, errors)
+
+
+def check_disallowed_characters(text):
+    # Anything that is not in this list: A-Za-z0-9.\s_- is banned
+    disallowed_characters_regex = r'[^\"A-Za-z0-9.\s_-]+'
```

Comment:
Just curious - why are whitespace `\s` and the quotation mark `"` included?

Reply:
I admit this is somewhat arbitrary and subjective, but I thought that "project name" is ok name, that someone could make without any mistake. Same for quotes, you can for example by mistake add prroject name with quotes inside scrapy.cfg. 

Creating the list of allowed characters is somewhat arbitrary, we could either go one extreme way and allow only digits and ascii, or go the other way and allow everything except things that are path separators. Path separators should be always disabled so that it stops someone running  scrapyd with sudo user from placing or deleting files anywhere in filesystem. 

I'm open for some feedback and suggestions here, maybe we can also check how other projects are doing things like this.

Reply:
The choice makes sense. I'm happy to go with it, as-is. You can maybe add a comment e.g. `Allow the same characters as Python packages, and allow project names like "my project".`